src/scipp/plt/model.py

```diff
@@ -0,0 +1,142 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from .widgets.widget import WidgetView
+
+from typing import Tuple, Iterable
+from functools import partial
+
+
+class Node:
+
+    def __init__(self, func, name=None, views=None):
+        self.name = name
+        self.parent_name = None
+        self.dependency = None
+        self.func = func
+        self.views = []
+        if views is not None:
+            for view in views:
+                self.views.append(view)
+
+    def send_notification(self, message):
+        # print(f'hey there! from {self.name}: ', message)
+        for view in self.views:
+            view.notify({
+                "node_name": self.name,
+                "parent_name": self.parent_name,
+                "message": message
+            })
+
+    def request_data(self):
+        pipeline = []
+        node = self
+        while node.dependency is not None:
+            pipeline.append(node.func)
+            node = node.dependency
+        else:
+            pipeline.append(node.func)
+        pipeline = list(pipeline[::-1])
+        data = pipeline[0]()
+        for step in pipeline[1:]:
+            data = step(data)
+        return data
+
+    def add_view(self, view):
+        self.views.append(view)
+
+
+class Model:
+
+    def __init__(self, da):
+        self._name = da.name
+        self._nodes = {}
+        self["root"] = Node(name='root', func=lambda: da)
+
+    def __getitem__(self, name: str) -> Node:
+        return self._nodes[name]
+
+    def __setitem__(self, name: str, node: Node):
+        self._nodes[name] = node
+        node.parent_name = self._name
+        node.name = name
+
+    def items(self) -> Iterable[Tuple[str, Node]]:
+        yield from self._nodes.items()
+
+    def children_of(self, name: str) -> Iterable[str]:
+        for node in self.values():
+            if node.dependency is not None:
+                if name == node.dependency.name:
+                    yield node
+
+    def keys(self) -> Iterable[str]:
+        yield from self._nodes.keys()
+
+    def values(self) -> Iterable[str]:
```

Comment:
Rename to `nodes`? Does this (and `keys`, `item`, `children_of`) need to be part of the public interface? I.e. how much do users (of this API) need to be aware of the fact that `Model` contains a graph and uses a dict to encode it?